Source/Bogus/DataSets/Date.cs

```diff
@@ -144,31 +129,221 @@ public DateTimeOffset FutureOffset(int yearsToGoForward = 1, DateTimeOffset? ref
 
-         return new DateTime(minTicks, start.Kind) + partTimeSpan;
+         var value = new DateTime(minTicks, DateTimeKind.Utc) + partTimeSpan;
+
+         if (start.Kind != DateTimeKind.Utc)
+         {
+            value = value.ToLocalTime();
+
+            if (value > end)
+               value = end;
+         }
+
+         return value;
+      }
+
+      /// <summary>
+      /// Takes a date/time range, as indicated by <paramref name="start"/> and <paramref name="end"/>,
+      /// and ensures that the range indicators are in the correct order and both reference actual
+      /// <see cref="DateTime"/> values. This takes into account the fact that when Daylight Savings Time
+      /// comes into effect, there is a 1-hour interval in the local calendar which does not exist, and
+      /// <see cref="DateTime"/> values in this change are not meaningful.
+      /// 
+      /// This function only worries about the start and end times. Impossible <see cref="DateTime"/>
+      /// values within the range are excluded automatically by means of the <see cref="DateTime.ToLocalTime"/>
+      /// function.
+      /// 
+      /// The version of this function built targeting .NET Standard 1.3 does not check Daylight Savings Time
+      /// transitions, as this API does not expose Daylight Savings Time information.
+      /// </summary>
+      /// <param name="start">A ref <see cref="DateTime"/> to be adjusted forward out of an impossible date/time range if necessary.</param>
+      /// <param name="end">A ref <see cref="DateTime"/> to be adjusted backward out of an impossible date/time range if necessary.</param>
+      /// <param name="kind">A <see cref="DateTimeKind"/> indicating how the supplied <see cref="DateTime"/> values should be interpreted with respect to DST change windows.</param>
+      /// <param name="preferRangeBoundary">An out <see cref="DateTime"/> that indicates which value should be used if the supplied range is empty due to being entirely contained within a DST transition range.</param>
+      private void ComputeRealRange(ref DateTime start, ref DateTime end, DateTimeKind kind, out DateTime preferRangeBoundary)
+      {
+         preferRangeBoundary = end;
+
+         if (start > end)
+         {
+            var tmp = start;
+
+            start = end;
+            end = tmp;
+         }
+
+#if !NETSTANDARD1_3
+         if (kind == DateTimeKind.Local)
+         {
+            var window = GetForwardDSTTransitionWindow(start);
+
+            var startLocal = start.ToLocalTime();
+            var endLocal = end.ToLocalTime();
+
+            if ((startLocal >= window.Start) && (startLocal <= window.End))
+            {
+               start = new DateTime(window.Start.Ticks, start.Kind);
+
+               if (start == end)
+                  return;
+            }
+
+            window = GetForwardDSTTransitionWindow(end);
+
+            if ((end >= window.Start) && (end < window.End))
```

Comment:
The DST transition window is the span of times that, in the local time zone, _don't exist_ because of the abrupt change do DST. But, `DateTime` comparisons don't take `Kind` into account and just look at the ticks. So, you can create `DateTime` values that point at the start and the end of the window that are _actually the same date/time_, but are numerically different. If you `.ToUtc()` them, you get the same UTC date/time out. This comparison is checking whether the desired end of the range from which to get a bogus DateTime lies within that nebulous region of date/time values that don't actually exist in the real world.